### PR TITLE
airgap: Remove airgap mode

### PR DESCRIPTION
**Pull request by kosstennbl**
## Description
Remove all code connected to airgap mode and module,
"offline" parameter, "input_file" and "output_file" arguments.
Related spec tests and documentation is also removed.

## Issues:
Refs: #2104

### CODE REVIEW

**Review by svteb — COMMENTED** [review]
Great job with the removal. I noticed a few mistakes that aren't exactly related to the change but we could fix them with this change.

File: src/tasks/cnf_setup.cr [hunk 1]
```diff
@@ -8,39 +8,16 @@ task "cnf_setup", ["helm_local_install", "create_namespace"] do |_, args|
...
-    release_name = CNFManager::Config.release_name_by_config_file(config_file)
-    if config_file && !AirGap.image_pull_policy_config_file?(install_method, config_src, release_name)
-      puts "Some containers within the installation manifests do not have an image pull policy defined.  Airgap mode will not work until this is fixed.".colorize(:red)
-      exit 1
-    end
-    AirGap.cache_images(cnf_setup: true)
-    puts "cnf setup finished caching images on nodes (airgapped mode)".colorize(:green)
-    CNFManager.sample_setup(cli_hash)
-    puts "cnf setup airgapped mode complete".colorize(:green)
+  Log.info { "Installing ClusterTools"}
```
[the remark this answers is not in the document]
Comment (kosstennbl) [reply src/tasks/cnf_setup.cr]: Agree, done

File: src/tasks/prereqs.cr [hunk 2]
```diff
@@ -8,25 +8,18 @@ require "helm"
...
-  offline_mode = false
-  offline_mode = true if args.named["offline"]?
-
   verbose = check_verbose(args)
   
   helm_condition = Helm::SystemInfo.helm_installation_info(verbose) && !Helm.helm_gives_k8s_warning?(true)
-
-  kubectl_existance = KubectlClient.installation_found?(verbose, offline_mode)
+  kubectl_existance = KubectlClient.installation_found?(verbose)
+  git_existance = GitClient.installation_found?
```
[the remark this answers is not in the document]
Comment (kosstennbl) [reply src/tasks/prereqs.cr]: Renamed to helm_ok, kubectl_ok and git_ok.

File: src/tasks/chaos_mesh_setup.cr [hunk 3]
```diff
@@ -3,30 +3,20 @@ require "file_utils"
...
-
-    if args.named["offline"]?
-      Log.info { "install chaos mesh offline mode" }
-      helm_chart = Dir.entries(CHAOS_MESH_OFFLINE_DIR).first
-      Helm.install("my-chaos-mesh #{CHAOS_MESH_OFFLINE_DIR}/#{helm_chart} --version 0.5.1")
-
-    else
-      # `helm repo add chaos-mesh https://charts.chaos-mesh.org`
-      # `helm install my-chaos-mesh chaos-mesh/chaos-mesh --version 0.5.1`
-      Helm.helm_repo_add("chaos-mesh","https://charts.chaos-mesh.org")
```
[the remark this answers is not in the document]
Comment (kosstennbl) [reply src/tasks/chaos_mesh_setup.cr:27]: It isn't removed, only taken from "else" branch to main code flow.

File: src/tasks/kubescape_setup.cr [hunk 4]
```diff
@@ -16,30 +16,18 @@ task "install_kubescape", ["kubescape_framework_download"] do |_, args|
...
-
-        stderr = IO::Memory.new
-        status = Process.run("chmod +x #{write_file}", shell: true, output: stderr, error: stderr)
-        success = status.success?
-        raise "Unable to make #{write_file} executable" if success == false
-      end
     end
 
-    # Irrespective of online or offline mode, write the version file with the kubescape version
+    # Write the version file with the kubescape version
```
[the remark this answers is not in the document]
Comment (kosstennbl) [reply src/tasks/kubescape_setup.cr:30]: Agree, code is self-explanatory

File: src/tasks/utils/cnf_manager.cr [hunk 5]
```diff
@@ -949,22 +866,8 @@ module CNFManager
...
         Log.for("verbose").info { "deploying with helm directory" } if verbose
-        # prepare a helm directory for deployment into an airgapped environment, put in airgap module
-        if input_file && !input_file.empty?
-          template_files = Dir.glob(["#{destination_cnf_dir}/#{helm_directory}/*.yaml*"])
-          template_files.map{|x| AirGap.image_pull_policy(x)}
-        end
         #TODO Add helm options into cnf-testsuite yml
         #e.g. helm install nsm --set insecure=true ./nsm/helm_chart
-        # Helm.template(release_name, install_method[1], output_file: "cnfs/temp_template.yml") 
-        Helm.template(release_name, install_method[1], output_file: "cnfs/temp_template.yml", namespace: helm_install_namespace, values: helm_values)
```
[the remark this answers is not in the document]
Comment (kosstennbl) [reply src/tasks/utils/cnf_manager.cr:960]: Should be, this code is needed only for image_pull, which was deleted completely.

**Review by martin-mat — APPROVED** [review]
lgtm

**Comment by Smitholi67** [comment]
Issue 2104 also makes reference to issue 1160.  Perhaps it should also be added on this PR?

**Comment by kosstennbl** [comment]
@Smitholi67 
At the time of creation of #1160, it wasn't decided to remove Airgap mode, the issue was about creating separate repo in https://github.com/cnf-testsuite and move code to there.

This PR deletes Airgap code completely (as was decided previously), and removes any support for this mode in the code. I don't see a reason to create additional repo for it.

**Review by Smitholi67 — APPROVED** [review]
lgtm

**Review by svteb — APPROVED** [review]

**Pull request merged**